### PR TITLE
Move incurred incidentials js to the right spot

### PR DESCRIPTION
**Pull request by tomecho**
* incurred incidentals logic should go into its own file
* also remove those imports which import already imported files, in application.js `require_tree .` imports all the files

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -25,14 +25,3 @@
 $(document).ready(function(){
```
Comment (tomecho) [inline app/assets/javascripts/application.js]: line 24 will import everything in `app/assets/javascripts`

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -1,66 +1,64 @@
...
-	<title>ProbableEngine</title>
-	<%= Gon::Base.render_data({}) %>
+  <head>
+    <title>ProbableEngine</title>
+    <%= Gon::Base.render_data({}) %>
 
-	<%= stylesheet_link_tag 'application', media: 'all' %>
-  <%= stylesheet_link_tag "#{params[:controller]}/#{params[:action]}" if File.exist?("#{Rails.root}/app/assets/stylesheets/#{params[:controller]}/#{params[:action]}.css") %>
-  <%= javascript_include_tag 'application' %>
-  <%= javascript_include_tag "#{params[:controller]}/#{params[:action]}" if File.exist?("#{Rails.root}/app/assets/javascripts/#{params[:controller]}/#{params[:action]}.js") %>
```
Comment (tomecho) [inline app/views/layouts/application.html.erb]: we're precompiling assets, we dont need to be including all assets and specific assets.

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -1,66 +1,64 @@
...
-  <%= csrf_meta_tags %>
-</head>
-<body>
-  <%= render partial: 'layouts/shared/header' %>
-  <!-- <%= render partial: 'layouts/shared/loading' %> -->
+    <%= csrf_meta_tags %>
+  </head>
+  <body>
+    <%= render partial: 'layouts/shared/header' %>
+    <!-- <%= render partial: 'layouts/shared/loading' %> -->
```
Comment (owlee) [inline app/views/layouts/application.html.erb]: Comment?

**Comment by tomecho** [comment]
What?

File: app/views/layouts/application.html.erb (hunk 3 above)
Comment (tomecho) [inline app/views/layouts/application.html.erb]: it looks like a retab

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (dfaulken) [inline app/views/layouts/application.html.erb]: "get me all the things and also this thing"

**Comment by tomecho** [comment]
[test](google.com)

**Pull request merged**